### PR TITLE
8514 - Load story template sections on scroll for C&U page only

### PR DESCRIPTION
**Pull request by anjenkin**
https://federal-spending-transparency.atlassian.net/browse/DA-8514

### CODE REVIEW

File: src/styles/progress.scss [hunk 1]
```diff
@@ -1,8 +1,46 @@
...
+
+
+@keyframes item {
+  from {background-color: #e8e8e8;}
+  to {background-color: #f8f8f8;}
 }
 
 .progress {
-  color: #512da8;
+  //color: #512da8;
```
Comment (brianpetway) [inline src/styles/progress.scss:45]: Can this be removed if it isn't needed?

File: src/styles/progress.scss [hunk 2]
```diff
@@ -1,8 +1,46 @@
...
+  animation-name: item;
+  animation-duration: 1s;
+  animation-iteration-count: infinite;
+}
+
+.body {
+  width: 100%;
+  border-radius: 5px;
+  margin-top: 10px;
+  height: 300px;
```
Comment (brianpetway) [inline src/styles/progress.scss:32]: There are two heights on this class.

File: src/styles/progress.scss [hunk 3]
```diff
@@ -1,8 +1,46 @@
...
+  animation-duration: 1s;
+  animation-iteration-count: infinite;
+}
+
+.teaser {
+  width: 100%;
+  height: 75px;
+  margin: 0 0 10px 0;
+  border-radius: 5px;
+  height: 100%;
```
Comment (brianpetway) [inline src/styles/progress.scss:22]: There are two heights on this class.

File: src/pages/colleges-and-universities/index.jsx [hunk 4]
```diff
@@ -104,11 +110,115 @@ const footnotesProp = [
...
+					}
+
+					const inView =
+						boundingRect.top < topThreshold &&
+						boundingRect.bottom > bottomThreshold;
+
+					if (entry.isIntersecting && ratio > 0.5 && ratio < 1 && inView) {
+						setActiveSectionIndex(sectionsByName.indexOf(section));
+
+						// this doesn't work for some reason
```
Comment (brianpetway) [inline src/pages/colleges-and-universities/index.jsx:186]: Is this comment still needed?

**Comment by brianpetway** [comment]
A few very small things noted inline above.
I can't tell what changes were made in mapbox.jsx bc most of the file is showing as changed bc of reformatting.
Section 2 and the mapbox component still seem to be taking a long time to load when you scroll down to them, but maybe it will be better in QAT.

File: src/pages/colleges-and-universities/index.jsx (hunk 4 above)
Comment (anjenkin) [reply src/pages/colleges-and-universities/index.jsx]: I update the comment to be more descriptive.  You can update the state here but the updated state isn't accessible from inside the intersection observer for some reason

File: src/styles/progress.scss (hunk 3 above)
Comment (anjenkin) [reply src/styles/progress.scss]: fixed

File: src/styles/progress.scss (hunk 1 above)
Comment (anjenkin) [reply src/styles/progress.scss]: this is actually necessary so added this back

**Comment by anjenkin** [comment]
Ready for re-review.  For mapbox, I removed lines 8 - 10 (the condition where we're checking if the props are not empty) and updated line 291 so the useEffect only runs once.  This was causing the map to reset

**Pull request merged**